server/app/views/applicant/NorthStarApplicantCommonIntakeUpsellView.java

```diff
@@ -0,0 +1,140 @@
+package views.applicant;
+
+import auth.CiviFormProfile;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import controllers.AssetsFinder;
+import controllers.LanguageUtils;
+import controllers.applicant.ApplicantRoutes;
+import java.util.ArrayList;
+import java.util.Locale;
+import modules.ThymeleafModule;
+import org.thymeleaf.TemplateEngine;
+import play.i18n.Messages;
+import play.mvc.Http.Request;
+import services.DeploymentType;
+import services.applicant.ApplicantPersonalInfo;
+import services.applicant.ApplicantService.ApplicantProgramData;
+import services.settings.SettingsManifest;
+
+public class NorthStarApplicantCommonIntakeUpsellView extends NorthStarApplicantBaseView {
+
+  @Inject
+  NorthStarApplicantCommonIntakeUpsellView(
+      TemplateEngine templateEngine,
+      ThymeleafModule.PlayThymeleafContextFactory playThymeleafContextFactory,
+      AssetsFinder assetsFinder,
+      ApplicantRoutes applicantRoutes,
+      SettingsManifest settingsManifest,
+      LanguageUtils languageUtils,
+      DeploymentType deploymentType) {
+    super(
+        templateEngine,
+        playThymeleafContextFactory,
+        assetsFinder,
+        applicantRoutes,
+        settingsManifest,
+        languageUtils,
+        deploymentType);
+  }
+
+  public String render(Params params) {
+    ThymeleafModule.PlayThymeleafContext context =
+        createThymeleafContext(
+            params.request(),
+            params.applicantId(),
+            params.profile(),
+            params.applicantPersonalInfo(),
+            params.messages());
+
+    context.setVariable("applicationId", params.applicationId());
+    context.setVariable("messages", params.messages());
+
+    String linkHref = settingsManifest.getCommonIntakeMoreResourcesLinkHref(params.request()).get();
+    String linkText = settingsManifest.getCommonIntakeMoreResourcesLinkText(params.request()).get();
+    String linkHtml =
+        "<a href=\""
+            + linkHref
+            + "\" target=\"_blank\" style=\"color: blue; text-decoration: underline;\">"
+            + linkText
+            + "</a>";
+    context.setVariable("moreResourcesLinkHtml", linkHtml);
+
+    ArrayList<DisplayProgram> displayPrograms = new ArrayList<DisplayProgram>();
+    Locale userLocale = params.messages().lang().toLocale();
```

Comment:
This could be simplified by using streams. Try something like (caveat that I wrote this in a github comment so there may be a compile error 😛 ):

`ImmutableList<DisplayProgram> displayPrograms = 
      params.eligiblePrograms().stream()
      .map(applicantProgramData -> 
          new DisplayProgram(applicantProgramData.program().localizedName().getOrDefault(userLocale), 
                                              applicantProgramData.program().localizedDescription().getOrDefault(userLocale))
     .collect(ImmutableList.toImmutableList());`